substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/aarch64/AArch64UContextRegisterDumper.java

```diff
@@ -50,7 +51,26 @@
 class AArch64UContextRegisterDumperFeature implements Feature {
     @Override
     public void afterRegistration(AfterRegistrationAccess access) {
-        ImageSingletons.add(UContextRegisterDumper.class, new AArch64UContextRegisterDumper());
+        UContextRegisterDumper regdumper;
+        if (SubstrateOptions.CompilerBackend.getValue().equals("llvm")) {
```

Comment:
@loicottet is this still necessary? Did we fix the segfault handler for AArch64 on LLVM?

Reply:
I think register dumping should work fine on ARM, there's no need to disable it for LLVM.

Reply:
It works on Linux AArch64, and it works on Darwin AMD64, but not on Darwin AArch64

Reply:
I guess the check should verify if we are running on Darwin for iOS (separate platform or not) instead of the LLVM backend then.

Reply:
Also, if it is not too hard I would simply implement this. Sooner or later there will be an issue that requires this dumper to be present: we had the same situation on DARWIN_AMD64 a few months ago.

Reply:
will do.

Reply:
I reverted the previous approach, and now have using DARWIN_AARCH64 the same as DARWIN_AMD64. The names of the context fields are similar between Darwin arm64 and amd64. I had to narrow the Platform to only LINUX_AARCH64 in the AArch64UContextRegisterDumper.